src/GovernanceToken.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts ^5.0.0
+pragma solidity ^0.8.20;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {ERC20Permit, Nonces} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Permit.sol";
+import {ERC20Votes} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
+
+contract GovernanceToken is ERC20, ERC20Permit, ERC20Votes {

```

Comment:
@PatrickAlphaC @Akash-Kolekar - Can one of you please explain why we do this double inheritance, I've seen it before in the Proxy Lesson...

If GovernanceToken is extending / inheriting from ERC20Votes which inherits from ERC20 what's the need for the explicit inheritance of the ERC20 contract by the GovernanceToken contract?

Reply:
It's a bit redundant yes! But we need it to tell solidity what functions we are overriding below.

Reply:
But doesn't it already know? At compile time shouldn't the solc
Compiler know this without the explicit inheritance in the Governance Token.